### ISSUE TITLE
Ability to use Rasa for internal bots and not only livechat

### ISSUE THREAD

**Issue by JargeZ** [RocketChat/Apps.Rasa#11]
Hello!

I had a need to write a bot for use inside a rocket chat. BUT the usual api does not allow using buttons to implement navigation.

I researched existing solutions, and with integration through the rocket chat application engine - there is only Rasa
However, as I figured out when trying to use, this only works for livechat.

Now I started writing an application that provides an interface for bots and I plan to make an adapter for an engine such as a botkit
However, I would also like to add the ability to work inside the chat for bots on the Rasa engine.

Therefore, I have the following questions:
- Did I understand correctly that now it only works with livechat?
- What do you think about if I try to implement support for internal chats and do a PR with this?

**Comment by murtaza98** [comment]
> Did I understand correctly that now it only works with livechat?

Yes, U r correct. This app is designed to work only with livechat / omnichannel

If u want to use Rasa for internal bots, I believe there is already a way you can accomplish this. Please refer [here](https://docs.rocket.chat/guides/bots/running-a-rasa-bot)

**Comment by JargeZ** [comment]
> I believe there is already a way you can accomplish this.

Yes, of course I used these guides and installed all the engines described in the documentation.
However, the biggest problem is the lack of any navigation that can be provided to the client. 
The adapters of the engines described above are often abandoned and run on a standard api, which does not allow rendering even simple navigation buttons =(

Now I started writing a universal extender as an application on the rocket engine, so that later I could use it in adapters for bots. I liked the hubot, but he is dead. 
Now I am looking towards Rasa and botkit from Microsoft

**Comment by xkungfu** [comment]
> > I believe there is already a way you can accomplish this.
> 
> Yes, of course I used these guides and installed all the engines described in the documentation.
> However, the biggest problem is the lack of any navigation that can be provided to the client.
> The adapters of the engines described above are often abandoned and run on a standard api, which does not allow rendering even simple navigation buttons =(
> 
> Now I started writing a universal extender as an application on the rocket engine, so that later I could use it in adapters for bots. I liked the hubot, but he is dead.
> Now I am looking towards Rasa and botkit from Microsoft

Hi JargeZ, same issue.
did you find a solution?

**Comment by JargeZ** [comment]
Hi @xkungfu 
i'm write [my own app](https://github.com/JargeZ/RocketChat.Apps.BotBridge) and [python adapter](https://github.com/JargeZ/RocketChat-Simple-AppBot) for simple bots. You can try using this, however it doesn't work with the Rasa at the moment.

Perhaps in the future I will write an adapter for a rasa bot for my application.
I will also see if it is possible to modify this Rasa.App

**Comment by xkungfu** [comment]
> Hi @xkungfu
> i'm write [my own app](https://github.com/JargeZ/RocketChat.Apps.BotBridge) and [python adapter](https://github.com/JargeZ/RocketChat-Simple-AppBot) for simple bots. You can try using this, however it doesn't work with the Rasa at the moment.
> 
> Perhaps in the future I will write an adapter for a rasa bot for my application.
> I will also see if it is possible to modify this Rasa.App

thanks. I will have a test soon.

why not simply use socketio as proxy server to pass message between rasa and rocket.chat? 

rasa server  <----> socketio server <-----> rocket.chat
